datasets/caner/caner.py

```diff
@@ -0,0 +1,133 @@
+@article{article,
+author = {Salah, Ramzi and Zakaria, Lailatul},
+year = {2018},
+month = {12},
+pages = {},
+title = {BUILDING THE CLASSICAL ARABIC NAMED ENTITY RECOGNITION CORPUS (CANERCORPUS)},
+volume = {96},
+journal = {Journal of Theoretical and Applied Information Technology}
+}
+"""
+
+_DESCRIPTION = """\
+Classical Arabic Named Entity Recognition corpus as a new corpus of tagged data that can be useful for handling the issues in recognition of Arabic named entities.
+"""
+
+_HOMEPAGE = "https://github.com/RamziSalah/Classical-Arabic-Named-Entity-Recognition-Corpus"
+
+# TODO: Add the licence for the dataset here if you can find it
+_LICENSE = ""
+
+_URL = "https://github.com/RamziSalah/Classical-Arabic-Named-Entity-Recognition-Corpus/archive/master.zip"
+
+
+class Caner(datasets.GeneratorBasedBuilder):
+    """Classical Arabic Named Entity Recognition corpus as a new corpus of tagged data that can be useful for handling the issues in recognition of Arabic named entities"""
+
+    VERSION = datasets.Version("1.1.0")
+
+    BUILDER_CONFIGS = [
+        datasets.BuilderConfig(name="first_domain", version=VERSION, description="This covers the training dataset"),
+    ]
+
+    DEFAULT_CONFIG_NAME = "first_domain"
+
+    def _info(self):
+
+        features = datasets.Features(
+            {
+                "tokens": datasets.Value("string"),
+                "ner_tags": datasets.ClassLabel(
+                    names=[
+                        "Allah",
+                        "Book",
+                        "Clan",
+                        "Crime",
+                        "Date",
+                        "Day",
+                        "Hell",
+                        "Loc",
+                        "Meas",
+                        "Mon",
+                        "Month",
+                        "NatOb",
+                        "Number",
+                        "O",
+                        "Org",
+                        "Para",
+                        "Pers",
+                        "Prophet",
+                        "Rlig",
+                        "Sect",
+                        "Time",
+                    ]
+                ),
+            }
+        )
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            supervised_keys=None,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+
+        my_urls = _URL
+        data_dir = dl_manager.download_and_extract(my_urls)
+
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                # These kwargs will be passed to _generate_examples
+                gen_kwargs={
+                    "filepath": os.path.join(
+                        data_dir, "Classical-Arabic-Named-Entity-Recognition-Corpus-master/CANERCorpus.csv"
```

Comment:
I noticed that in the CANER github repository there're also many txt files.
What's the difference between the txt files and this csv ?

I'm asking because in this csv there's one row per token but it seems that we can't reconstruct the sentences. In general for NER datasets we try to have sentences instead of individual tokens (see [conll2003](https://github.com/huggingface/datasets/blob/master/datasets/conll2003/conll2003.py) for example, in which each sample contains several tokens)

If there's no way to get the sentences we should stick with the individual tokens though, as it is right now.

Reply:
Good spot, although I had realised that we try to have sentences instead of individual tokens but when I read through the paper and looked through all the datasets (in txt and csv versions) none had any indications as to when a sentence starts or ends unfortunately.

Reply:
Also reduced the dummy_data.zip file as requested.